src/cma.ts

```diff
@@ -34,146 +53,53 @@ async function generateCMASpawnArguments(command) {
   return [`${adapterDir}/cma_bin/cma`, [command]];
 }
 
-
 /**
  * Invoke the cumulus-message-adapter
  *
  * @returns {Promise<Object>} cumulusMessageAdapterObject - Returns an Object with a
  * childprocess and a stderror buffer
  * @returns {Object} cumulusMessageAdapterObject.cmaProcess - A CMA childProcess Object
- * @returns {string} cumulusMessageAdapterObject.errorObj -  A Object with the property
+ * @returns {Object} cumulusMessageAdapterObject.errorObj -  A Object with the property
  *                                                           'stderrBuffer' to make the encapsulated
  *                                                           error event storage outside this method
  */
-async function invokeCumulusMessageAdapter() {
+export async function invokeCumulusMessageAdapter(): Promise<InvokeCumulusMessageAdapterResult> {
   const spawnArguments = await generateCMASpawnArguments('stream');
   const errorObj = { stderrBuffer: '' };
   try {
-    // Would like to use sindresorhus's lib, however
-    // https://github.com/sindresorhus/execa/issues/411
-    // and related mean that pulling in the childProcess
-    // is hacky.   Using node native for now, should revisit
-    // the results #414 in the future.
     const cmaProcess = childProcess.spawn(...spawnArguments);
-    cmaProcess.on('error', () => {});
-    cmaProcess.stdin.setEncoding = 'utf8';
-    cmaProcess.stdout.setEncoding = 'utf8';
-    cmaProcess.stderr.setEncoding = 'utf8';
+    cmaProcess.on('error', () => { });
+    cmaProcess.stdin.setDefaultEncoding('utf8');
+    cmaProcess.stdout.setEncoding('utf8');
+    cmaProcess.stderr.setEncoding('utf8');
     cmaProcess.on('close', () => {
+      // eslint-disable-next-line no-console
```

Comment:
Given the purpose of this package, I'd probably just disable this rule for the entire package.

Reply:
I can live with that: e3687e9